src/services/course.js

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable import/prefer-default-export */
+import { Op } from 'sequelize';
+import parse from 'csv-parse/lib/sync';
+import db, { sequelize } from '../models';
+
+const {
+  Course,
+} = db;
+
+export async function csvImport(buffer) {
+  const created = [];
+  const replaced = [];
+  const updated = [];
+  const deleted = [];
+  const errors = [];
+  const skipped = [];
+
+  const importedCourseIds = [];
+
+  const parsed = parse(buffer, { skipEmptyLines: true, columns: true });
+  let rowCount = 1;
+  const results = parsed.map(async (course) => {
```

Comment:
Since your loading the whole file anyway, why are you processing it one line at a time. when the steps could be performed in batches to reduce the distinct calls to sql and increase performance and decrease the duration of the lock being placed on all the tables involved.

Reply:
I don't see this file being very large, this is also how we handle it in other places.